File: src/main/kotlin/lec08/Lec08Main.java
package lec08;

public class Lec08Main {
    public static void main(String[] args) {
        repeat("2", 1, true);
        repeat("2", 1);
        repeat("2", false);
    }

    public static void repeat(final String str, final int num, final boolean useNewLine) {
        for (int i = 0; i <= num; i++)
            if (useNewLine) System.out.println(str);
            else System.out.print(str);
    }

    public static void repeat(final String str, final int num) {
        repeat(str, num, true);
    }

    public static void repeat(final String str, final boolean useNewLine) {
        repeat(str, 13, useNewLine);
    }
}

File: src/main/kotlin/lec08/Lec08Main.kt
package lec08

fun main() {
    println(max(1, 2))
    repeat("Hello, World!", useNewLine = false)
}

fun max(a: Int, b: Int) = if (a > b) a else b

fun repeat(str: String, num: Int = 3, useNewLine: Boolean = true): Unit {
    for (i in 1..num) if (useNewLine) println(str) else print(str)
}
